### PR TITLE
update containerd binary to v1.6.1

### PR DESCRIPTION
**Pull request by thaJeztah**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by samuelkarp — APPROVED** [review]

File: Dockerfile.windows [hunk 1]
```diff
@@ -166,7 +166,7 @@ FROM microsoft/windowsservercore
 SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
 
 ARG GO_VERSION=1.17.3
-ARG CONTAINERD_VERSION=1.5.7
+ARG CONTAINERD_VERSION=1.6.0-beta.2
```
Comment (thaJeztah) [inline Dockerfile.windows:169]: Note to self; add temporary fix for version string / download URL in Windows Dockerfile

**Comment by thaJeztah** [comment]
Argh.. I never understand PowerShell;

```
https://github.com/containerd/containerd/releases/download/v${Env:CONTAINERD_VERSION}/containerd-${Env:CONTAINERD_VERSION}-windows-amd64.tar.gz :
The term 'https://github.com/containerd/containerd/releases/download/v${Env:CONTAINERD_VERSION}/containerd-${Env:CONTAINERD_VERSION}-windows-amd64.tar.gz'
is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify
that the path is correct and try again. At line:1 char:3159
```

Guess it's just back to concatenating stuff `<flip-table />`

**Comment by tianon** [comment]
```powershell
$location='https://github.com/containerd/containerd/releases/download/v'+$Env:CONTAINERD_VERSION+'/containerd-'+$Env:CONTAINERD_VERSION+'-windows-amd64.tar.gz';
```

Could instead be:

```powershell
$location = 'https://github.com/containerd/containerd/releases/download/v{0}/containerd-{0}-windows-amd64.tar.gz' -f $env:CONTAINERD_VERSION;
```

Demo:

```console
PS C:\> $location = 'https://github.com/containerd/containerd/releases/download/v{0}/containerd-{0}-windows-amd64.tar.gz' -f $env:CONTAINERD_VERSION 
PS C:\> write-host $location
https://github.com/containerd/containerd/releases/download/v1.6.0-beta.4/containerd-1.6.0-beta.4-windows-amd64.tar.gz
```

**Comment by thaJeztah** [comment]
Thanks! So I tried putting the `${Env:Foo}` inside the string, which felt like a slightly more readable approach;

```powershell
$location="https://github.com/containerd/containerd/releases/download/v${Env:CONTAINERD_VERSION}/containerd-${Env:CONTAINERD_VERSION}-windows-amd64.tar.gz"
```


Which (looking at https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-string-substitutions?view=powershell-7.2) _seemed_ like it was supported, but not sure why it then tries to _execute_ the string.

Guess the devil is in the detail ... somewhere.

Not worth spending too much time on for now, I guess, so I just reverted it to what it was

**Comment by thaJeztah** [comment]
Maybe it expects it to be used without `Env:` and I should've used the plain `$VAR` or `${VAR}` in it, idk 😂

**Comment by thaJeztah** [comment]
Failing tests:

- :warning: `TestRunContainerWithRmFlagCannotStartContainer` (possible regression, but could be racy)
- :warning: `TestRunMutableNetworkFiles` (possible regression)
- :warning: `TestDaemonRestartSaveContainerExitCode` (possible regression)
- :heavy_check_mark: (likely flaky) `TestNetworkDBIslands`
- :heavy_check_mark: (likely flaky) `TestNetworkDBCRUDTableEntries`
- :heavy_check_mark: (likely flaky) `TestJSONFileLoggerWithOpts`

-------------

This test looks like a possible regression, although it was known to be racy before (see https://github.com/moby/moby/issues/11966)

```
=== RUN   TestDockerSuite/TestRunContainerWithRmFlagCannotStartContainer
    docker_cli_run_test.go:2776: Expected not to have containers 2492808e26cc

    --- FAIL: TestDockerSuite/TestRunContainerWithRmFlagCannotStartContainer (0.43s)
```

This could be a regression; although I can find various mentions of this test, some of which indicate it may have been racy https://github.com/moby/moby/search?p=1&q=TestRunMutableNetworkFiles&type=issues
"exit status 1" looks fishy though

```
=== RUN   TestDockerSuite/TestRunMutableNetworkFiles
    docker_cli_exec_test.go:389: assertion failed:
        Command:  /usr/local/cli/docker rm -fv 2492808e26cc
        ExitCode: 1
        Error:    exit status 1
        Stdout:
        Stderr:   Error response from daemon: No such container: 2492808e26cc


        Failures:
        ExitCode was 1 expected 0
        Expected no error
    --- FAIL: TestDockerSuite/TestRunMutableNetworkFiles (0.04s)
```

This could be a regression (don't see many mentions of that test failing, and online from a long time ago) https://github.com/moby/moby/search?q=TestDaemonRestartSaveContainerExitCode&type=issues

```
=== RUN   TestDockerDaemonSuite/TestDaemonRestartSaveContainerExitCode
    docker_cli_daemon_test.go:2636: assertion failed: 137 (out string) != 127 (string)
    --- FAIL: TestDockerDaemonSuite/TestDaemonRestartSaveContainerExitCode (1.46s)
```


Failure in `TestNetworkDBIslands` - known flaky: https://github.com/moby/moby/issues/42459


Windows 2022 / containerd (flaky test); https://github.com/moby/moby/issues/42739

```
=== RUN   TestNetworkDBCRUDTableEntries
    networkdb_test.go:310: Entry existence verification test failed for node2(1f082a212f1b)
2021/12/22 18:01:59 Closing DB instances...
--- FAIL: TestNetworkDBCRUDTableEntries (7.91s)
```

Windows RS5 (flaky test?) was flaky before: https://github.com/moby/moby/issues/36801

```
=== RUN   TestJSONFileLoggerWithOpts
    jsonfilelog_test.go:187: open C:\Users\ContainerAdministrator\AppData\Local\Temp\docker-logger-2690863916\container.log.1: The process cannot access the file because it is being used by another process.
--- FAIL: TestJSONFileLoggerWithOpts (0.02s)
```

**Comment by thaJeztah** [comment]
New run of CI:

On amd64: this is starting to look like a regression:

```
=== RUN   TestDockerDaemonSuite/TestDaemonRestartSaveContainerExitCode
    docker_cli_daemon_test.go:2636: assertion failed: 137 (out string) != 127 (string)
    --- FAIL: TestDockerDaemonSuite/TestDaemonRestartSaveContainerExitCode (1.74s)
```

On Windows RS5: known flaky test https://github.com/moby/moby/issues/38521

```
=== RUN   TestDockerSuite/TestStartReturnCorrectExitCode
    docker_cli_start_test.go:209: assertion failed: expected an error, got nil
    --- FAIL: TestDockerSuite/TestStartReturnCorrectExitCode (9.84s)
```

Giving it another run

**Comment by tianon** [comment]
Running `docker run --rm busybox toto` and even `docker run --rm --init busybox toto` by hand with Docker 20.10 or a fresh build from master combined with https://github.com/containerd/containerd/releases/download/v1.6.0-beta.5/containerd-1.6.0-beta.5-linux-amd64.tar.gz works every time and returns 127 as expected. :weary:

**Comment by thaJeztah** [comment]
@tianon thanks for trying! So 🤔 differences in our CI situation would be;

- it's running docker-in-docker
- it's integration-cli, and using the docker 17.06 cli (although for these options, I can't see how that would matter)


For easier finding; this is the test that's failing; https://github.com/moby/moby/blob/0f1d65b2e0cd0cb2e7c2b8855cf8b6f624d9b052/integration-cli/docker_cli_daemon_test.go#L2556

And it _looks_ pretty straightforward (run a container that uses a non-existing binary as command); https://github.com/moby/moby/blob/0f1d65b2e0cd0cb2e7c2b8855cf8b6f624d9b052/integration-cli/docker_cli_daemon_test.go#L2568-L2575

**Comment by thaJeztah** [comment]
oh, and of course the CI uses the frozen images, so the version of the busybox image is https://github.com/moby/moby/blob/0f1d65b2e0cd0cb2e7c2b8855cf8b6f624d9b052/Dockerfile#L91-L92

**Comment by tianon** [comment]
Looks like the server is using 20.10.10-rc1 -- any idea what version of `libseccomp`? (Just to start ruling things out.)

Maybe we can try running that specific integration test locally to see if we can reproduce outside CI too (or making a test commit that limits CI to _just_ that test and see if that passes :joy: :see_no_evil:).

**Comment by thaJeztah** [comment]
oh, good call on the daemon version on the host 🤔

guess we should update those again (need to find the right repository again for that 🤣)

**Comment by thaJeztah** [comment]
I'm able to reproduce the issue when running inside the dev container (`make BIND_DIR=. binary shell`);

```bash
docker pull busybox

docker run --name error-values --init=false busybox toto
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "toto": executable file not found in $PATH: unknown.
ERRO[0004] error waiting for container: context canceled

docker inspect -f '{{.State.ExitCode}}' error-values
137
```

daemon logs related to the above:

<details>

```
DEBU[2022-01-10T11:55:43.507894777Z] Calling POST /v1.30/containers/create?name=error-values
DEBU[2022-01-10T11:55:43.509033292Z] form data: {"AttachStderr":true,"AttachStdin":false,"AttachStdout":true,"Cmd":["toto"],"Domainname":"","Entrypoint":null,"Env":[],"HostConfig":{"AutoRemove":false,"Binds":null,"BlkioDeviceReadBps":null,"BlkioDeviceReadIOps":null,"BlkioDeviceWriteBps":null,"BlkioDeviceWriteIOps":null,"BlkioWeight":0,"BlkioWeightDevice":null,"CapAdd":null,"CapDrop":null,"Cgroup":"","CgroupParent":"","ConsoleSize":[0,0],"ContainerIDFile":"","CpuCount":0,"CpuPercent":0,"CpuPeriod":0,"CpuQuota":0,"CpuRealtimePeriod":0,"CpuRealtimeRuntime":0,"CpuShares":0,"CpusetCpus":"","CpusetMems":"","DeviceCgroupRules":null,"Devices":[],"DiskQuota":0,"Dns":[],"DnsOptions":[],"DnsSearch":[],"ExtraHosts":null,"GroupAdd":null,"IOMaximumBandwidth":0,"IOMaximumIOps":0,"Init":false,"IpcMode":"","Isolation":"","KernelMemory":0,"Links":null,"LogConfig":{"Config":{},"Type":""},"Memory":0,"MemoryReservation":0,"MemorySwap":0,"MemorySwappiness":-1,"NanoCpus":0,"NetworkMode":"default","OomKillDisable":false,"OomScoreAdj":0,"PidMode":"","PidsLimit":0,"PortBindings":{},"Privileged":false,"PublishAllPorts":false,"ReadonlyRootfs":false,"RestartPolicy":{"MaximumRetryCount":0,"Name":"no"},"SecurityOpt":null,"ShmSize":0,"UTSMode":"","Ulimits":null,"UsernsMode":"","VolumeDriver":"","VolumesFrom":null},"Hostname":"","Image":"busybox","Labels":{},"NetworkingConfig":{"EndpointsConfig":{}},"OnBuild":null,"OpenStdin":false,"StdinOnce":false,"Tty":false,"User":"","Volumes":{},"WorkingDir":""}
DEBU[2022-01-10T11:55:43.530752968Z] container mounted via layerStore: &{/var/lib/docker/overlay2/826c78df23cd3bc99913e9d0f78426fe15e9b93e4c5ae683f92842a39482952c/merged 0x344f560 0x344f560}  container=086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81
DEBU[2022-01-10T11:55:43.548080697Z] Calling POST /v1.30/containers/086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81/attach?stderr=1&stdout=1&stream=1
DEBU[2022-01-10T11:55:43.548626728Z] attach: stdout: begin
DEBU[2022-01-10T11:55:43.548974635Z] attach: stderr: begin
DEBU[2022-01-10T11:55:43.549582742Z] Calling POST /v1.30/containers/086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81/wait?condition=next-exit
DEBU[2022-01-10T11:55:43.550842848Z] Calling POST /v1.30/containers/086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81/start
DEBU[2022-01-10T11:55:43.552067723Z] container mounted via layerStore: &{/var/lib/docker/overlay2/826c78df23cd3bc99913e9d0f78426fe15e9b93e4c5ae683f92842a39482952c/merged 0x344f560 0x344f560}  container=086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81
DEBU[2022-01-10T11:55:43.552912809Z] Assigning addresses for endpoint error-values's interface on network bridge
DEBU[2022-01-10T11:55:43.552973651Z] RequestAddress(LocalDefault/172.18.0.0/16, <nil>, map[])
DEBU[2022-01-10T11:55:43.553026259Z] Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65533, Sequence: (0xc0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:<nil>
DEBU[2022-01-10T11:55:43.561313886Z] Assigning addresses for endpoint error-values's interface on network bridge
DEBU[2022-01-10T11:55:43.565904466Z] Programming external connectivity on endpoint error-values (8bc07793c0be5196b6c6c8fac9d662826486503d84c9a393e180c9769959f016)
DEBU[2022-01-10T11:55:43.566957234Z] EnableService 086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 START
DEBU[2022-01-10T11:55:43.567003158Z] EnableService 086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 DONE
DEBU[2022-01-10T11:55:43.570363138Z] bundle dir created                            bundle=/var/run/docker/containerd/086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 module=libcontainerd namespace=moby root=/var/lib/docker/overlay2/826c78df23cd3bc99913e9d0f78426fe15e9b93e4c5ae683f92842a39482952c/merged
DEBU[2022-01-10T11:55:43.577312870Z] event published                               ns=moby topic=/containers/create type=containerd.events.ContainerCreate
time="2022-01-10T11:55:43.599712129Z" level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
time="2022-01-10T11:55:43.599950679Z" level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
time="2022-01-10T11:55:43.600201167Z" level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
time="2022-01-10T11:55:43.601266058Z" level=info msg="starting signal loop" namespace=moby path=/run/docker/containerd/daemon/io.containerd.runtime.v2.task/moby/086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 pid=944 runtime=io.containerd.runc.v2
DEBU[2022-01-10T11:55:43.648009404Z] garbage collected                             d=1.142049ms
DEBU[2022-01-10T11:55:43.756585282Z] sandbox set key processing took 65.709707ms for container 086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81
DEBU[2022-01-10T11:55:43.785690280Z] failed to delete task                         error="rpc error: code = NotFound desc = container not created: not found" id=086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81
INFO[2022-01-10T11:55:43.786308808Z] shim disconnected                             id=086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81
WARN[2022-01-10T11:55:43.786391840Z] cleaning up after shim disconnected           id=086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 namespace=moby
INFO[2022-01-10T11:55:43.786465203Z] cleaning up dead shim
WARN[2022-01-10T11:55:43.802849319Z] cleanup warnings time="2022-01-10T11:55:43Z" level=info msg="starting signal loop" namespace=moby pid=985 runtime=io.containerd.runc.v2
DEBU[2022-01-10T11:55:43.803506166Z] event published                               ns=moby topic=/tasks/exit type=containerd.events.TaskExit
DEBU[2022-01-10T11:55:43.803658180Z] event published                               ns=moby topic=/tasks/delete type=containerd.events.TaskDelete
ERRO[2022-01-10T11:55:43.803804251Z] copy shim log                                 error="read /proc/self/fd/13: file already closed"
DEBU[2022-01-10T11:55:43.804853618Z] event                                         module=libcontainerd namespace=moby topic=/tasks/exit
DEBU[2022-01-10T11:55:43.804921764Z] event                                         module=libcontainerd namespace=moby topic=/tasks/delete
ERRO[2022-01-10T11:55:43.804998376Z] stream copy error: reading from a closed fifo
INFO[2022-01-10T11:55:43.806326148Z] ignoring event                                container=086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
ERRO[2022-01-10T11:55:43.805060257Z] stream copy error: reading from a closed fifo
DEBU[2022-01-10T11:55:43.805399937Z] attach: stderr: end
DEBU[2022-01-10T11:55:43.805395759Z] attach: stdout: end
DEBU[2022-01-10T11:55:43.807399980Z] attach done
DEBU[2022-01-10T11:55:43.809764528Z] event published                               ns=moby topic=/containers/delete type=containerd.events.ContainerDelete
DEBU[2022-01-10T11:55:43.817078809Z] Revoking external connectivity on endpoint error-values (8bc07793c0be5196b6c6c8fac9d662826486503d84c9a393e180c9769959f016)
DEBU[2022-01-10T11:55:43.817712158Z] DeleteConntrackEntries purged ipv4:0, ipv6:0
DEBU[2022-01-10T11:55:43.836560091Z] garbage collected                             d=1.266253ms
DEBU[2022-01-10T11:55:43.872421803Z] Releasing addresses for endpoint error-values's interface on network bridge
DEBU[2022-01-10T11:55:43.872493585Z] ReleaseAddress(LocalDefault/172.18.0.0/16, 172.18.0.2)
DEBU[2022-01-10T11:55:43.872521696Z] Released address PoolID:LocalDefault/172.18.0.0/16, Address:172.18.0.2 Sequence:App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65532, Sequence: (0xe0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:3
ERRO[2022-01-10T11:55:43.882144245Z] 086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 cleanup: failed to delete container from containerd: no such container
WARN[2022-01-10T11:55:43.883828105Z] error locating sandbox id 8e2aff22ecad17b08fe07d5dfdd2f73544c6bc46c1c5bcd1d1e04cc205db5a35: sandbox 8e2aff22ecad17b08fe07d5dfdd2f73544c6bc46c1c5bcd1d1e04cc205db5a35 not found
DEBU[2022-01-10T11:55:43.883930433Z] Failed to unmount 826c78df23cd3bc99913e9d0f78426fe15e9b93e4c5ae683f92842a39482952c overlay: /var/lib/docker/overlay2/826c78df23cd3bc99913e9d0f78426fe15e9b93e4c5ae683f92842a39482952c/merged - no such file or directory  storage-driver=overlay2
ERRO[2022-01-10T11:55:43.884643229Z] 086678ef8b6507782b2cfe88160b89cc43958a438910dcaf9e1352c989d5fd81 cleanup: failed to delete container from containerd: no such container
```

</details>

**Comment by thaJeztah** [comment]
I cleaned up logs for both the 1.6.0 and 1.5.8 situation (replacing timestamps, container ID's etc.) and did a compare between the two:

[containerd-1.5.8.log.txt](https://github.com/moby/moby/files/7838725/containerd-1.5.8.log.txt)
[containerd-1.6.0.log.txt](https://github.com/moby/moby/files/7838726/containerd-1.6.0.log.txt)

Diff between those:

```patch
diff --git a/containerd-1.5.8.log b/containerd-1.6.0.log
index 73f052e5e6..ca156f0821 100644
--- a/containerd-1.5.8.log
+++ b/containerd-1.6.0.log
@@ -1,4 +1,3 @@
-DEBU[xxxx] Calling GET /_ping
 DEBU[xxxx] Calling POST /v1.30/containers/create?name=error-values
 DEBU[xxxx] form data: {"AttachStderr":true,"AttachStdin":false,"AttachStdout":true,"Cmd":["toto"],"Domainname":"","Entrypoint":null,"Env":[],"HostConfig":{"AutoRemove":false,"Binds":null,"BlkioDeviceReadBps":null,"BlkioDeviceReadIOps":null,"BlkioDeviceWriteBps":null,"BlkioDeviceWriteIOps":null,"BlkioWeight":0,"BlkioWeightDevice":null,"CapAdd":null,"CapDrop":null,"Cgroup":"","CgroupParent":"","ConsoleSize":[0,0],"ContainerIDFile":"","CpuCount":0,"CpuPercent":0,"CpuPeriod":0,"CpuQuota":0,"CpuRealtimePeriod":0,"CpuRealtimeRuntime":0,"CpuShares":0,"CpusetCpus":"","CpusetMems":"","DeviceCgroupRules":null,"Devices":[],"DiskQuota":0,"Dns":[],"DnsOptions":[],"DnsSearch":[],"ExtraHosts":null,"GroupAdd":null,"IOMaximumBandwidth":0,"IOMaximumIOps":0,"Init":false,"IpcMode":"","Isolation":"","KernelMemory":0,"Links":null,"LogConfig":{"Config":{},"Type":""},"Memory":0,"MemoryReservation":0,"MemorySwap":0,"MemorySwappiness":-1,"NanoCpus":0,"NetworkMode":"default","OomKillDisable":false,"OomScoreAdj":0,"PidMode":"","PidsLimit":0,"PortBindings":{},"Privileged":false,"PublishAllPorts":false,"ReadonlyRootfs":false,"RestartPolicy":{"MaximumRetryCount":0,"Name":"no"},"SecurityOpt":null,"ShmSize":0,"UTSMode":"","Ulimits":null,"UsernsMode":"","VolumeDriver":"","VolumesFrom":null},"Hostname":"","Image":"busybox","Labels":{},"NetworkingConfig":{"EndpointsConfig":{}},"OnBuild":null,"OpenStdin":false,"StdinOnce":false,"Tty":false,"User":"","Volumes":{},"WorkingDir":""}
 DEBU[xxxx] container mounted via layerStore: &{/var/lib/docker/overlay2/yyyy/merged 0x344f560 0x344f560}  container=yyyy
@@ -17,16 +16,24 @@ DEBU[xxxx] EnableService yyyy START
 DEBU[xxxx] EnableService yyyy DONE
 DEBU[xxxx] bundle dir created                            bundle=/var/run/docker/containerd/yyyy module=libcontainerd namespace=moby root=/var/lib/docker/overlay2/yyyy/merged
 DEBU[xxxx] event published                               ns=moby topic=/containers/create type=containerd.events.ContainerCreate
-time="xxxx" level=info msg="starting signal loop" namespace=moby path=/run/docker/containerd/daemon/io.containerd.runtime.v2.task/moby/yyyy pid=659
+time="xxxx" level=info msg="loading plugin \"io.containerd.event.v1.publisher\"..." runtime=io.containerd.runc.v2 type=io.containerd.event.v1
+time="xxxx" level=info msg="loading plugin \"io.containerd.internal.v1.shutdown\"..." runtime=io.containerd.runc.v2 type=io.containerd.internal.v1
+time="xxxx" level=info msg="loading plugin \"io.containerd.ttrpc.v1.task\"..." runtime=io.containerd.runc.v2 type=io.containerd.ttrpc.v1
+time="xxxx" level=info msg="starting signal loop" namespace=moby path=/run/docker/containerd/daemon/io.containerd.runtime.v2.task/moby/yyyy pid=944 runtime=io.containerd.runc.v2
 DEBU[xxxx] garbage collected                             d=1.xxxxms
 DEBU[xxxx] sandbox set key processing took 62.xxxxms for container yyyy
 DEBU[xxxx] failed to delete task                         error="rpc error: code = NotFound desc = container not created: not found" id=yyyy
 INFO[xxxx] shim disconnected                             id=yyyy
 WARN[xxxx] cleaning up after shim disconnected           id=yyyy namespace=moby
 INFO[xxxx] cleaning up dead shim
-WARN[xxxx] cleanup warnings time="xxxx" level=info msg="starting signal loop" namespace=moby pid=xxxx
+WARN[xxxx] cleanup warnings time="xxxx" level=info msg="starting signal loop" namespace=moby pid=xxxx runtime=io.containerd.runc.v2
+DEBU[xxxx] event published                               ns=moby topic=/tasks/exit type=containerd.events.TaskExit
+DEBU[xxxx] event published                               ns=moby topic=/tasks/delete type=containerd.events.TaskDelete
 ERRO[xxxx] copy shim log                                 error="read /proc/self/fd/13: file already closed"
+DEBU[xxxx] event                                         module=libcontainerd namespace=moby topic=/tasks/exit
+DEBU[xxxx] event                                         module=libcontainerd namespace=moby topic=/tasks/delete
 ERRO[xxxx] stream copy error: reading from a closed fifo
+INFO[xxxx] ignoring event                                container=yyyy module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
 ERRO[xxxx] stream copy error: reading from a closed fifo
 DEBU[xxxx] attach: stdout: end
 DEBU[xxxx] attach: stderr: end
@@ -39,3 +46,6 @@ DEBU[xxxx] Releasing addresses for endpoint error-values's interface on network
 DEBU[xxxx] ReleaseAddress(LocalDefault/172.18.0.0/16, 172.18.0.2)
 DEBU[xxxx] Released address PoolID:LocalDefault/172.18.0.0/16, Address:172.18.0.2 Sequence:App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65532, Sequence: (0xe0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:3
 ERRO[xxxx] yyyy cleanup: failed to delete container from containerd: no such container
+WARN[xxxx] error locating sandbox id yyyy: sandbox yyyy not found
+DEBU[xxxx] Failed to unmount yyyy overlay: /var/lib/docker/overlay2/yyyy/merged - no such file or directory  storage-driver=overlay2
+ERRO[xxxx] yyyy cleanup: failed to delete container from containerd: no such container
```

**Comment by Mohsen7s** [comment]
containerd 1.6.0 already released yesterday which it solve a lot of bugs with experimental checkpoint feature.
Can we have finally ship containerd 1.6.0 ?

**Comment by AkihiroSuda** [comment]
Could you rebase and update to v1.6.1?
LGTM then

**Comment by tianon** [comment]
Given @AkihiroSuda showing me what I was doing wrong in https://github.com/containerd/containerd/issues/6659#issuecomment-1064812512 (thanks again! :heart:), I'm +1 on updating to 1.6.1 also.

**Comment by thaJeztah** [comment]
Updated to v1.6.1

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
@cpuguy83 @tonistiigi @AkihiroSuda we all ok with going to v1.6 for the next release?

**Review by AkihiroSuda — APPROVED** [review]

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**